tests/test_file_open_gui.py

```diff
@@ -0,0 +1,449 @@
+# Copyright (c) 2020 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import pytest
+import subprocess
+import time
+import os
+import sys
+import sgtk
+from tk_toolchain.authentication import get_toolkit_user
+
+try:
+    from MA.UI import topwindows
+except ImportError:
+    pytestmark = pytest.mark.skip()
+
+
+@pytest.fixture(scope="session")
+def context():
+    # Tasks in Toolkit File Open UI Automation project which we're going to use
+    # in different test cases.
+    # Get credentials from TK_TOOLCHAIN
+    sg = get_toolkit_user().create_sg_connection()
+
+    # Create or update the integration_tests local storage with the current test run
+    storage_name = "File Open UI Tests"
+    local_storage = sg.find_one(
+        "LocalStorage", [["code", "is", storage_name]], ["code"]
+    )
+    if local_storage is None:
+        local_storage = sg.create("LocalStorage", {"code": storage_name})
+
+    # Always update local storage path
+    local_storage["path"] = os.path.expandvars("${SHOTGUN_CURRENT_REPO_ROOT}")
+    sg.update(
+        "LocalStorage", local_storage["id"], {"windows_path": local_storage["path"]}
+    )
+
+    # Make sure there is not already an automation project created
+    filters = [["name", "is", "Toolkit File Open UI Automation"]]
+    existed_project = sg.find_one("Project", filters)
+    if existed_project is not None:
+        sg.delete(existed_project["type"], existed_project["id"])
+
+    # Create a new project with the Film VFX Template
+    project_data = {
+        "sg_description": "Project Created by Automation",
+        "name": "Toolkit File Open UI Automation",
+    }
+    new_project = sg.create("Project", project_data)
+
+    # Create a Sequence to be used by the Shot creation
+    sequence_data = {
+        "project": {"type": new_project["type"], "id": new_project["id"]},
+        "code": "seq_001",
+        "sg_status_list": "ip",
+    }
+    new_sequence = sg.create("Sequence", sequence_data)
+
+    # Create a new shot
+    shot_data = {
+        "project": {"type": new_project["type"], "id": new_project["id"]},
+        "sg_sequence": {"type": new_sequence["type"], "id": new_sequence["id"]},
+        "code": "shot_001",
+        "sg_status_list": "ip",
+        "task_template": {"type": "TaskTemplate", "id": 5},
```

Comment:
We should look for that task template by it's name and not just 5.

Reply:
Yeah I know but the thing is, I looked at different Shotgun site and it turns out that depending on the unified template version used when the site gets created, task templates can have different names but they always have the same id.

Reply:
Now using custom task templates

Reply:
Now using custom task templates